airflow/utils/db.py

```diff
@@ -1018,8 +1018,42 @@ def upgradedb(session: Session = NEW_SESSION):
         exit(1)
 
     with create_global_lock(session=session, lock=DBLocks.MIGRATIONS):
-        log.info("Creating tables")
-        command.upgrade(config, 'heads')
+        log.info("Running migrations")
+        rev_2_0_0_head = 'e959f08ac86c'
+        if ':' in revision:
+            sql = True
+            _lowerband, _upperband = revision.split(':')
+            head_to_lowerband_range = f"{rev_2_0_0_head}:{_lowerband}"
+            head_to_upperband_range = f"{rev_2_0_0_head}:{_upperband}"
+            for i in [head_to_lowerband_range, head_to_upperband_range]:
+                try:
+                    command.history(config, rev_range=i)
+                except Exception:
+                    raise AirflowException(
+                        f"Error while checking history for revision range {i}. "
+                        f"Check that {i.split(':')[1]} is a valid revision. "
+                        f"Supported revision for offline migration is from {rev_2_0_0_head} "
+                        f"which is airflow 2.0.0 head"
+                    )
+            log.info("Running offline migrations for revision range %s", revision)
+        elif sql:
+            # user supplied a specific SQL revision
+            revision = f"{rev_2_0_0_head}:{revision}"
+            try:
+                command.history(config, rev_range=revision)
```

Comment:
Doesn't this produce output? Is that intended?

Reply:
It produces output and I can't think of a way to silence it. There's no other way I could confirm that we should be good at producing the migration

Reply:
Working on suppressing the logs..

Reply:
I couldn't suppress the log. I think there's no problem if the log is there. Users would copy from BEGIN TRANSACTION. Apart from the history log, alembic also logs before the SQL